x-pack/plugins/cases/public/components/all_cases/index.test.tsx

```diff
@@ -926,4 +926,25 @@ describe('AllCasesGeneric', () => {
       ).toBeFalsy();
     });
   });
+
+  it('should not render status when isSelectorView=true', async () => {
+    const wrapper = mount(
+      <TestProviders>
+        <AllCases {...defaultAllCasesProps} isSelectorView={true} />
+      </TestProviders>
+    );
+
+    const { result } = renderHook<GetCasesColumn, CasesColumns[]>(() =>
+      useCasesColumns({
+        ...defaultColumnArgs,
+        isSelectorView: true,
+      })
+    );
+
+    expect(result.current.find((i) => i.name === 'Status')).toBeFalsy();
+
+    await waitFor(() => {
+      expect(wrapper.find('[data-test-subj="case-view-status-dropdown"]').exists()).toBeFalsy();
+    });
```

Comment:
Question: Is this `waitFor` waiting for the dropdown not to be there? If the modal is not open yet when it is tested, won't it pass the test anyway? I am not sure, but if it's the case, maybe we could "waitFor" the modal to exist, and then test that the dropdown is not there.

Reply:
The `waitFor` is mostly for the various hooks of the component. If I do not put it within the `waitFor` then I get the following warning:

```
 Warning: An update to CasesTableFiltersComponent inside a test was not wrapped in act(...).

    When testing, code that causes React state updates should be wrapped into act(...):

    act(() => {
      /* fire events that update state */
    });
    /* assert on the output */

    This ensures that you're testing the behavior the user would see in the browser. Learn more at https://fb.me/react-wrap-tests-with-act
        in CasesTableFiltersComponent
        in span
        in Unknown
        in OwnerProvider (created by TestProvidersComponent)
        in ThemeProvider (created by TestProvidersComponent)
        in Provider
        in Unknown (created by TestProvidersComponent)
        in PseudoLocaleWrapper (created by I18nProvider)
        in IntlProvider (created by I18nProvider)
        in I18nProvider (created by TestProvidersComponent)
        in TestProvidersComponent (created by WrapperComponent)
        in WrapperComponent
```

> If the modal is not open yet when it is tested, won't it pass the test anyway? I am not sure, but if it's the case, maybe we could "waitFor" the modal to exist, and then test that the dropdown is not there.

The component that we test is not modal. It is put into a body of a modal by another component. In this component, if you pass the `isSelectorView` prop you tell the component that it will be used to select cases not to show them. You are right that we should wait for the component to be ready and then check the existence of the dropdown. I changed it accordingly. Thanks!